crates/node-db/tests/insert.rs

```diff
@@ -216,15 +224,18 @@ fn test_failed_block() {
     // Create an in-memory SQLite database
     let mut conn = test_conn();
 
-    // Create the necessary tables and insert the block.
-    let tx = conn.transaction().unwrap();
-    node_db::create_tables(&tx).unwrap();
-    for block in &blocks {
-        node_db::insert_block(&tx, block).unwrap();
-    }
+    let mut r = vec![];
+    node_db::with_tx::<_, QueryError>(&mut conn, |tx| {
+        // Create the necessary tables and insert the block.
+        node_db::create_tables(tx).unwrap();
+        for block in &blocks {
+            node_db::insert_block(tx, block).unwrap();
+        }
+        r = node_db::list_blocks(tx, 0..(NUM_BLOCKS + 10)).unwrap();
```

Comment:
Similar to above, we should return the result of `list_blocks` rather than assigning to a mutable var